lib/lz4hc.c

```diff
@@ -287,6 +303,37 @@ LZ4HC_InsertAndGetWiderMatch (
         }   }   }   }
     }  /* while ((matchIndex>=lowLimit) && (nbAttempts)) */
 
+    if (dict == usingDictCtx && nbAttempts && ipIndex - lowLimit < MAX_DISTANCE) {
+        size_t const dictEndOffset = dictCtx->end - dictCtx->base;
+        assert(dictEndOffset <= 1 GB);
+        dictMatchIndex = dictCtx->hashTable[LZ4HC_hashPtr(ip)];
+        matchIndex = dictMatchIndex + lowLimit - (U32)dictEndOffset;
+        while (ipIndex - matchIndex <= MAX_DISTANCE && nbAttempts--) {
+            const BYTE* const matchPtr = dictCtx->base + dictMatchIndex;
+
+            if (LZ4_read32(matchPtr) == pattern) {
+                int mlt;
+                int back = 0;
+                const BYTE* vLimit = ip + (dictCtx->end - matchPtr);
```

Comment:
minor, more food for thought : 
`dictCtx->end - matchPtr == dictEndOffset - dictMatchIndex`.

The differences are : 
- 2nd calculation is done in `U32` space, rather than as a pointer difference. Ptr difference may more easily invoke undefined behaviors (though, in this case, it should completely fine, since both `end` and `matchPtr` point to the same object).
- arguments for 2nd calculation are available before `matchPtr` is calculated. In theory it's better for performance, but in this case, I don't expect it to make any tangible benefit : there are tons of mis-predictions in this code, which cost a lot more.